controllers/machineset_controller_test.go

```diff
@@ -244,6 +248,10 @@ func TestMachineSetReconciler(t *testing.T) {
 			g.Expect(im.GetAnnotations()).To(HaveKeyWithValue("annotation-1", "true"), "have annotations of MachineTemplate applied")
 			g.Expect(im.GetAnnotations()).To(HaveKeyWithValue("precedence", "MachineSet"), "the annotations from the MachineSpec template to overwrite the bootstrap config template ones")
 			g.Expect(im.GetLabels()).To(HaveKeyWithValue("label-1", "true"), "have labels of MachineTemplate applied")
+			g.Expect(im.GetOwnerReferences()).To(HaveLen(2))
+			for _, o := range im.GetOwnerReferences() {
```

Comment:
Same than https://github.com/kubernetes-sigs/cluster-api/pull/5865/files#r769007709

Reply:
see https://github.com/kubernetes-sigs/cluster-api/pull/5865#discussion_r769020994
(resolving because it is a duplicate)